miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultInstrumentService.java

```diff
@@ -71,9 +77,74 @@ public long create(Instrument instrument) throws IOException {
     return save(instrument);
   }
 
+  @Override
+  public void addServiceRecord(ServiceRecord record, Instrument instrument) throws IOException {
+    // originally located at HibernateServiceRecordDao
+    if (instrument.getDateCommissioned() != null) {
+      throw new IOException("Cannot add service records to a retired instrument!");
+    }
+    long recordId = serviceRecordService.create(record);
+    ServiceRecord serviceRecord = serviceRecordService.get(recordId);
+
+    // add saved service record
+    instrument.getServiceRecords().add(serviceRecord);
+
+    // save instrument
+    save(instrument);
+  }
+
+  @Override
+  public void removeServiceRecord(ServiceRecord record, Instrument instrument) throws IOException {
+    if (!instrument.getServiceRecords().contains(record)) {
+      throw new IOException("Cannot remove service record that does not exist");
+    }
+    instrument.getServiceRecords().remove(record);
+    save(instrument);
+
+    // remove servicerecord
+    serviceRecordService.delete(record);
+  }
+
+  @Override
+  public void updateServiceRecord(ServiceRecord record, Instrument instrument) throws IOException {
+
+    if (!instrument.getServiceRecords().contains(record)) {
```

Comment:
Please do these operations on `managedInstrument`, and throw `ValidationException` here in the service